SimpleGame/README.md

```diff
@@ -0,0 +1,18 @@
+I wrote this program as an exercise to explore some concepts for the Sequence Executor engine. The program is an incomplete implementation of a simple game, where a ball moves inside a box at constant speed, changing direction in response to user inputs. The ball position update is driven by a simulated periodic timer. The periodic timer and the user inputs are modeled as streams of data.
+The concepts I wanted to test are presented below. 
+# Model the basic execution unit as a state machine that produces status updates.
+The execution of a sequence can be modeled as a state machine that produces a stream of status updates (I’m using the term status update to refer to a snapshot of the sequence execution state, with the information relevant to the clients). The output stream will ultimately feed the web server to send updates to the clients. The state machine is constructed as a chain of smaller state machines. Each one takes care of an action in the sequence and produces a status update.
+A second related idea is to compose the state that is passed around with the sequence execution state and the sequence commands (just the ones that affect a running sequence: `stop`, `setBreakpoint`). The action state machine will then be composed of two stages. The first one gets the input commands from the real world and updates the commands part of the state. The second one executes the action according to the previous state and the commands. For example, if the commands section has `stop==true`, then the action part should end the execution and close the output stream. That structure isolates the interaction with the real world from the rest of the processing. Part of this idea is implemented in _Game.scala_, where the state machine created by `updateSpeed` takes a player input and updates the speed direction in the state. The updated state is then passed to the state machine created by the function `tick`, which provides a new state with an updated position.
+# Time control everywhere
+Another concept I want to use is to treat time as an input (an idea I first read about in John Carmack’s .plan). To achieve this I need to:
```

Comment:
Someone put all those .plan files in Github. Here is the one about time: https://raw.githubusercontent.com/ESWAT/john-carmack-plan-archive/master/by_day/johnc_plan_19981014.txt